src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -624,86 +623,95 @@ public enum CompareExchangeTombstonesCleanupState
             InvalidPeriodicBackupStatus,
             NoMoreTombstones
         }
-
-        private CompareExchangeTombstonesCleanupState GetMaxCompareExchangeTombstonesEtagToDelete<TRavenTransaction>(TransactionOperationContext<TRavenTransaction> context, string databaseName, DatabaseObservationState state, out long maxEtag) where TRavenTransaction : RavenTransaction
+        
+        private CompareExchangeTombstonesCleanupState GetMaxCompareExchangeTombstonesEtagToDelete<TRavenTransaction>(TransactionOperationContext<TRavenTransaction> context, string databaseName, MergedDatabaseObservationState mergedState, out long maxEtag) where TRavenTransaction : RavenTransaction
         {
             List<long> periodicBackupTaskIds;
             maxEtag = long.MaxValue;
 
-            if (state?.RawDatabase != null)
+            if (mergedState.RawDatabase != null)
             {
-                periodicBackupTaskIds = state.RawDatabase.PeriodicBackupsTaskIds;
+                periodicBackupTaskIds = mergedState.RawDatabase.PeriodicBackupsTaskIds;
             }
             else
             {
                 using (var rawRecord = _server.Cluster.ReadRawDatabaseRecord(context, databaseName))
                     periodicBackupTaskIds = rawRecord.PeriodicBackupsTaskIds;
             }
 
-            if (periodicBackupTaskIds != null && periodicBackupTaskIds.Count > 0)
+            foreach (var (shardNumber, state) in mergedState.States)
```

Comment:
what this will yield for a non-sharded database?

Reply:
maybe create a method that will accept `databaseName`, `state` &  `periodicBackupTaskIds` ?

Reply:
`mergedState.States` will contain the non-sharded db state.